ui/src/pages/passwordReset/SendEmail.tsx

```diff
@@ -0,0 +1,62 @@
+import { auth } from '@lib/firebase';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { sendResetPasswordEmailSchema } from 'consts/validationSchemas';
+import { fetchSignInMethodsForEmail } from '@firebase/auth';
+import { sendPasswordResetEmail } from 'firebase/auth';
+import { FirebaseError } from '@firebase/util';
+import { useMutation } from '@tanstack/react-query';
+import LoadingScreen from '@components/LoadingScreen';
+import { NavLink } from 'react-router-dom';
+import paths from '@routes/paths';
+import AuthenticateLayout from 'layouts/AuthenticateLayout';
+import Form from '@components/Form';
+import { t } from 'i18next';
+import Input from '@components/Input';
+import Button from '@components/Button';
+
+interface ResetPasswordEmail {
+  email: string;
+}
+
+const SendEmail = () => {
+  const methods = useForm<ResetPasswordEmail>({
+    resolver: yupResolver(sendResetPasswordEmailSchema),
+  });
+
+  const findUserAndSendResetPasswordMail = async (email: string) => {
+    const user = await fetchSignInMethodsForEmail(auth, email);
+    if (user.length > 0) return sendPasswordResetEmail(auth, email);
+    else return Promise.reject(new FirebaseError('auth/user-not-found', 'Firebase: Error (auth/user-not-found).'));
+  };
+
+  const { mutate, isSuccess, isPending, isError, error } = useMutation<void, FirebaseError, ResetPasswordEmail>({
+    mutationFn: (values: ResetPasswordEmail) => findUserAndSendResetPasswordMail(values.email),
+  });
+
```

Comment:
podsumowując te dwa komentarze to warto zrobić np coś takiego:
* stworzyć folder api 
* w tym folderze tworzyć foldery na podstawie wykonywanych akcji, tutaj mogłoby to być np auth
* w folderze auth umieścic plik requests i w nim funkcję `findUserAndSendResetPasswordMail`
* w folderze auth umieśccić plik hooks i w nim hook `useResetPasswordMutation` który zwraca `useMutation<void, FirebaseError, ResetPasswordEmail>({
    mutationFn: (values: ResetPasswordEmail) => findUserAndSendResetPasswordMail(values.email),
  });`

Dzięki temu mamy reużywalne hooki których możemy użyć w kilku miejscach, dodatkowo jeśli chodzi o użycie react-query możemy łatwo zarządzać kluczami bo wszystko mamy w jednym miejscu i wiemy kiedy jaki klucz będzie trzeba unieważnić jeśli np wykonamy najpierw zapytanie post i potem bedziemy chcieli pobrać nowe dane